content/articles/mern-isp-management-system/index.md

```diff
@@ -0,0 +1,611 @@
+
+`npm install cors`
+
+We will install nodemon to run our server automatically when we make changes.
+
+`npm install nodemon --save-dev`
+
+We have to change the scripts in package.json to 
+`"devStart": "nodemon index.js"`
+We will also install dotenv to handle our variables from a .env file.
+
+`npm install dotenv --save-dev`
+
+Note that nodemon and dotenv are in our development dependencies.
+
+### Setting up the Server
+
+Create a .env file with the following code.
+
+```
+DATABASE_URL = mongodb://localhost/customers
+PORT = 5000
+```
+Let us create index.js and add the following code.
+
+```JavaScript
+require('dotenv').config()
+const cors = require('cors')
+const express = require('express')
+const app = express()
+const mongoose = require('mongoose')
+const PORT = process.env.PORT || 5010;
+
+app.use(express.json())
+app.use(cors())
+
+app.listen(PORT, () => console.log(`Server Started on port ${PORT}`))
+
+```
+
+First, we import all the libraries we installed.
+We define the app variable as an instance of express, which we will use to configure our server.
+
+app. use() allows us to use the different libraries needed, including `app. use(express.json())`, which is a middleware function that parses incoming JSON requests and puts the parsed data in req. body.
+app. listen()listens to the connections on the specified port and executes the callback function we pass.
+process.env allows us to access environment variables 
+
+To establish a connection to the database, add the following code:
+
+```Javascript
+mongoose.connect(process.env.DATABASE_URL,
+    {
+        useNewUrlParser: true,
+        useUnifiedTopology: true
+    })
+
+const db = mongoose.connection
+db.on('error', (error) => console.error(error))
+db.once('open', () => console.log('Connected to Database'))
+```
+
+Now let us define the database models. Create a models folder and inside the folder, create a Customer.js file which will contain our customer schema.
+
+```JavaScript
+const mongoose = require('mongoose')
+
+const customerSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: true
+    },
+    email:{
+        type: String,
+        required: true
+    },
+    phoneNumber: {
+        type: Number,
+        required: true,
+    },
+    location: {
+        type: String,
+        required: true,
+
+    },
+    subscribedToPackage: {
+        type: String,
+        required: true
+    },
+    subscribeDate: {
+        type: Date,
+        required: true,
+        default: Date.now
+    }
+})
+
+module.exports = mongoose.model('Customer', customerSchema)
+
+```
+The schema contains the customer's properties which are defined as a JavaScript object where we specify the data type and whether a field is required.
+
+
+### Setting up API Endpoints
+
+We need to set up the API endpoints to handle our HTTP requests. We are interested in the POST, GET, PATCH and DELETE methods.
```

Comment:
Why we need these API endpoints? State the use of each and relate it to our problem statement.